server/controllers/clubProfileController.js

```diff
@@ -82,23 +84,37 @@ module.exports = {
     },
     create: function(req, res) {
         // TODO; req.body contains profile information
-        clubProfile.create(req.body)
-                .then(newclubProfile => res.json(newclubProfile))
+        clubProfile.create(req.body) 
+                .then(newclubProfile => {
+                  //set new profile's acc ID to the given accID
+                  newclubProfile.clubAccountId = req.params.accountId;
+                  //get the account assosiated with given accID and set its profile Id to the new club profile Id
+                  var newClubAccount = getById(req.params.accountId,res);
+                  newClubAccount.clubProfileId = newclubProfile.id;
+                  res.json(newclubProfile)
+                })
                 .catch(err => res.status(422).json(err))
     },
     update: function(req, res) {
         // TODO; req.body contains profile information and req.params.id
-        clubProfile.findOneAndUpdate({ _id: req.params.id}, req.body)
+        // req.params.accountId is account id 
+        // find associated profile from this
+        var updateProfile = getProfile(req.params.accountId,res); 
+        clubProfile.findOneAndUpdate({ _id: updateProfile.id}, req.body) 
                 .then(clubprofile => res.json(clubprofile))
                 .catch(err => res.status(422).json(err))
     },
     delete: function(req, res) {
         // TODO: req.params.id
-        clubProfile.findById({ _id: req.params.id })
-                .then(clubprofile => clubprofile.remove())
+        var deleteAccount = getById(req.params.accountId);
+        deleteAccount.clubProfileId = null;
+        var deleteProfile = getProfile(req.params.accountId ,res); 
+        clubProfile.findById({ _id: deleteProfile.id })
+                .then(clubprofile => {
+                  clubprofile.remove()
+                })
                 .then(allprofiles => res.json(allprofiles))
-                .catch(err => res.status(422).json(err))
-                
+                .catch(err => res.status(422).json(err))   
```

Comment:
given the above comments, this could be done in a simpler way utilizing `findOneAndDelete`. You can take a look at how the `delete` function is currently implemented in `clubAccountControllers` (remember to pull from `DBMaster` first!)